File: server/Models/Soba.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace server.Models
{
    [Table("Sobe")]
    public class Soba
    {
        [Key]
        [Column("ID")]
        public int ID { get; set; }

        [Column("BrojKreveta")]
        public int BrojKreveta { get; set; }
        
        public virtual List<Krevet> Kreveti { get; set; }
        
        [JsonIgnore]
        public Bolnica bolnica { get; set; }
    }
}

File: client/soba.js
import { Krevet } from "./krevet.js";

export class Soba{


    constructor(id, brojKreveta) {
        this.id = id
        this.kreveti = [];
    }

    dodajKrevet(krevet)
    {
        this.kreveti.push(krevet);
    }

    vratiBrojKreveta()
    {
        return this.kreveti.length;
    }

    daLiImaSlobodnihMesta()
    {
        for(let i=0; i<this.kreveti.length; i++)
            if(this.kreveti[i].slobodan())
                return true;
        return false;
    }

}